native/java/org/jpype/pkg/JPypePackage.java

```diff
@@ -0,0 +1,126 @@
+package org.jpype.pkg;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Map;
+import org.jpype.JPypeKeywords;
+
+public class JPypePackage
+{
+
+  final String pkg;
+  final Map<String, Path> contents;
+
+  public JPypePackage(String pkg, Map<String, Path> contents)
+  {
+    this.pkg = pkg;
+    this.contents = contents;
+  }
+
+  public Object getObject(String name)
+  {
+    Path p = contents.get(name);
+    if (p == null)
+      return null;
+    if (Files.isDirectory(p))
+      return p.toString().replace("/", ".");
+    if (p.toString().endsWith(".class"))
+    {
+      try
+      {
+        if (!isPublic(p))
+          return null;
+        return Class.forName(pkg + "." + JPypeKeywords.unwrap(name));
+      } catch (ClassNotFoundException ex)
+      {
+        return null;
+      }
+    }
+    return null;
+  }
+
+  public String[] getContents()
+  {
+    ArrayList<String> out = new ArrayList<>();
+    for (String key : contents.keySet())
+    {
+      Object o = this.getObject(key);
+      if (o != null)
+        out.add(key);
+    }
+    return out.toArray(new String[out.size()]);
+  }
+
+  static boolean isPublic(Path p)
```

Comment:
the method returns true if the path contains a public package/class? A short docstring wouldn't hurt. The magic going on here is fascinating :)

Reply:
Done.

Reply:
Fascinating because it looks useful, or fascinating because I spent multiple weeks digging through technical documents for over 3 years figuring it out?  😪